capture/project_library/project.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Optional
+
+import chipwhisperer as cw
+import numpy as np
+
+sys.path.append("../")
+from capture.project_library.ot_trace_library.trace_library import (  # noqa: E402
+    Trace, TraceLibrary)
+
+
+@dataclass
+class ProjectConfig:
+    """ Project configuration.
+    Stores information about the project.
+    """
+    type: str
+    path: Path
+    wave_dtype: np.dtype
+    overwrite: bool
+    trace_threshold: Optional[int] = 1
+
+
+class SCAProject:
+    """ Project class.
+
+    Used to manage a SCA project.
+    """
+    def __init__(self, project_cfg: ProjectConfig) -> None:
+        self.project_cfg = project_cfg
+        self.project = None
+
+    def create_project(self):
+        """ Create project.
+
+        Create project using the provided path.
+        """
+        project = None
+        if self.project_cfg.type == "cw":
+            project = cw.create_project(self.project_cfg.path,
+                                        overwrite=self.project_cfg.overwrite)
+        elif self.project_cfg.type == "ot_trace_library":
```

Comment:
nit: do we want an error msg if no valid selected, or will this just explode bc we then need to check all other cfg params too? :)

Reply:
Added an error msg. Config path is sanitized by the argument parser, but the other configs are currently not checked.